### PR TITLE
chore: add Git Hook to validate commit messages

### PR DESCRIPTION
**Pull request by rauliyohmc**

### CODE REVIEW

File: scripts/validateCommitMessage.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+const fs = require('fs');
```
Comment (satya164) [inline scripts/validateCommitMessage.js]: Use Flow?

File: scripts/validateCommitMessage.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+const TYPES = {
+  feat: true,
+  fix: true,
+  docs: true,
+  style: true,
+  refactor: true,
+  perf: true,
+  test: true,
+  chore: true,
+  revert: true,
```
Comment (satya164) [inline scripts/validateCommitMessage.js]: breaking too

File: scripts/validateCommitMessage.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  return true;
+}
+
+function firstLineFromBuffer(buffer) {
+  return buffer.toString().split('\n').shift();
+}
+
+const commitMsgFile = process.argv[2];
+
+fs.readFile(commitMsgFile, function(err, buffer) {
```
Comment (satya164) [inline scripts/validateCommitMessage.js]: maybe `fs.readFileSync` is better since it's a script

Comment (rauliyohmc) [inline scripts/validateCommitMessage.js]: Yeah, changed

File: scripts/validateCommitMessage.js (hunk 2 above)
Comment (rauliyohmc) [inline scripts/validateCommitMessage.js]: Added

File: scripts/validateCommitMessage.js (hunk 1 above)
Comment (rauliyohmc) [inline scripts/validateCommitMessage.js]: Should I? There are only 2 functions to annotate, one with `string` type and `buffer` type, which is a node object

Comment (satya164) [inline scripts/validateCommitMessage.js]: @rauliyohmc flow provides autocomplete

Comment (rauliyohmc) [inline scripts/validateCommitMessage.js]: Cool, added

**Pull request merged**